plugins/org.jkiss.dbeaver.ext.hana/src/org/jkiss/dbeaver/ext/hana/model/HANAStructureAssistant.java

```diff
@@ -115,8 +117,11 @@ public List<DBSObjectReference> findObjectsByMask(@NotNull DBRProgressMonitor mo
                 (GenericSchema) params.getParentObject() : (params.isGlobalSearch() || !(executionContext instanceof GenericExecutionContext) ? null : ((GenericExecutionContext) executionContext).getDefaultSchema());
 
         try (JDBCSession session = executionContext.openSession(monitor, DBCExecutionPurpose.META, "Find objects by mask")) {
-            if (objectTypesList.contains(HANAObjectType.TABLE) || objectTypesList.contains(HANAObjectType.VIEW) || objectTypesList.contains(HANAObjectType.PROCEDURE) ||
-                    objectTypesList.contains(HANAObjectType.SYNONYM)) {
+            if (objectTypesList.contains(HANAObjectType.TABLE)

```

Comment:
1. Do not convert array params.getObjectTypes() into list
2. Use ArrayUtils.containsAny and ArrayUtils.contains instead of huge || || ||.